packages/@react-spectrum/button/src/Button.tsx

```diff
@@ -43,12 +47,22 @@ function Button(props: SpectrumButtonProps, ref: FocusableRef) {
   let {buttonProps, isPressed} = useButton(props, domRef);
   let {hoverProps, isHovered} = useHover({isDisabled});
   let {styleProps} = useStyleProps(otherProps);
+  let {isIndeterminate} = useIndeterminate({isPending});
+  let progressCircleProps: SpectrumProgressCircleProps = {
+    isIndeterminate: true,
+    size: 'S',
+    'aria-hidden': true,
+    ...variant === 'overBackground' && {variant: 'overBackground'}
+  };
 
   let buttonVariant = variant;
   if (VARIANT_MAPPING[variant]) {
     buttonVariant = VARIANT_MAPPING[variant];
   }
 
+  // TODO: preserve width of hidden children, append localized "loading"
+  children = isIndeterminate ? <VisuallyHidden>{children}</VisuallyHidden> : children;
```

Comment:
If you add a en-US file you can copy "loading" from Picker. See i18n folder in Picker.